### PR TITLE
WFLY-12722 Upgrade Infinispan to 10.1.8.Final

### PR DESCRIPTION
**Pull request by pferraro**
https://issues.redhat.com/browse/WFLY-12722

Requires https://github.com/wildfly/wildfly/pull/13282

10.0.0.Final: https://issues.redhat.com/projects/ISPN/versions/12332894
10.1.0.Final: https://issues.redhat.com/projects/ISPN/versions/12342474
10.1.1.Final: https://issues.redhat.com/projects/ISPN/versions/12344239
10.1.2.Final: https://issues.redhat.com/projects/ISPN/versions/12344558
10.1.3.Final: https://issues.redhat.com/projects/ISPN/versions/12344717
10.1.4.Final: https://issues.redhat.com/projects/ISPN/versions/12344975
10.1.5.Final: https://issues.redhat.com/projects/ISPN/versions/12345160
10.1.6.Final: https://issues.redhat.com/projects/ISPN/versions/12345252
10.1.7.Final: https://issues.redhat.com/projects/ISPN/versions/12345475
10.1.8.Final: https://issues.redhat.com/projects/ISPN/versions/12345671

### CODE REVIEW

**Comment by pferraro** [comment]
@rhusar Please review.

**Comment by rhusar** [comment]
@pferraro Needs a rebase.

**Comment by rhusar** [comment]
@pferraro Needs a rebase.

**Comment by rhusar** [comment]
@pferraro Needs a rebase now.

**Comment by rhusar** [comment]
retest this please

File: clustering/infinispan/extension/src/main/resources/schema/jboss-as-infinispan_10_0.xsd [hunk 1]
```diff
@@ -0,0 +1,1467 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
+<!--
+  ~ JBoss, Home of Professional Open Source.
+  ~ Copyright 2019, Red Hat, Inc., and individual contributors
```
Comment (rhusar) [inline clustering/infinispan/extension/src/main/resources/schema/jboss-as-infinispan_10_0.xsd:5]: Lets to 2020 here.

Comment (pferraro) [reply clustering/infinispan/extension/src/main/resources/schema/jboss-as-infinispan_10_0.xsd:5]: :)

File: clustering/infinispan/extension/src/main/resources/schema/jboss-as-infinispan_10_0.xsd [hunk 2]
```diff
@@ -0,0 +1,1467 @@
...
+            <xs:annotation>
+                <xs:documentation>
+                    The maximum size of a batch to be inserted/deleted from the store.
+                    If the value is less than one, then no upper limit is placed on the number of operations in a batch.
+                </xs:documentation>
+            </xs:annotation>
+        </xs:attribute>
+        <xs:attribute name="shared" type="xs:boolean" default="false">
+            <xs:annotation>
+                <xs:documentation>This setting should be set to true when multiple cache instances share the same cache store (e.g., multiple nodes in a cluster using a JDBC-based CacheStore pointing to the same, shared database.) Setting this to true avoids multiple cache instances writing the same modification multiple times. If enabled, only the node where the modification originated will write to the cache store. If disabled, each individual cache reacts to a potential remote update by storing the data to the cache store.</xs:documentation>
```
Comment (rhusar) [inline clustering/infinispan/extension/src/main/resources/schema/jboss-as-infinispan_10_0.xsd:498]: Unrelated to this PR I would prefer if we could break this >550 char line. Its not rendered too well on GitHub.

**Review by rhusar — CHANGES_REQUESTED** [review]

File: clustering/infinispan/extension/src/main/resources/schema/jboss-as-infinispan_10_0.xsd [hunk 3]
```diff
@@ -0,0 +1,1499 @@
...
+            </xs:enumeration>
+        </xs:restriction>
+    </xs:simpleType>
+
+    <xs:simpleType name="protocol-version">
+        <xs:restriction base="xs:token">
+            <xs:enumeration value="1.0"/>
+            <xs:enumeration value="1.1"/>
+            <xs:enumeration value="1.2"/>
+            <xs:enumeration value="1.3"/>
```
Comment (rhusar) [inline clustering/infinispan/extension/src/main/resources/schema/jboss-as-infinispan_10_0.xsd:1438]: Versions prior to 2.0 have been dropped:

https://github.com/infinispan/infinispan/blob/10.1.7.Final/client/hotrod-client/src/main/java/org/infinispan/client/hotrod/ProtocolVersion.java#L27

File: clustering/infinispan/extension/src/main/resources/schema/jboss-as-infinispan_10_0.xsd [hunk 4]
```diff
@@ -0,0 +1,1499 @@
...
+            </xs:annotation>
+        </xs:attribute>
+        <xs:attribute name="name" type="xs:string" use="required">
+            <xs:annotation>
+                <xs:documentation>
+                    Uniquely identifies this remote cache container.
+                </xs:documentation>
+            </xs:annotation>
+        </xs:attribute>
+        <xs:attribute name="protocol-version" type="tns:protocol-version" default="2.6">
```
Comment (rhusar) [inline clustering/infinispan/extension/src/main/resources/schema/jboss-as-infinispan_10_0.xsd:1280]: The default version appears to be now 3.0.

https://github.com/infinispan/infinispan/blob/10.1.7.Final/client/hotrod-client/src/main/java/org/infinispan/client/hotrod/ProtocolVersion.java#L41

**Review by rhusar — CHANGES_REQUESTED** [review]

File: clustering/infinispan/extension/src/main/resources/schema/jboss-as-infinispan_10_0.xsd (hunk 4 above)
Comment (pferraro) [reply clustering/infinispan/extension/src/main/resources/schema/jboss-as-infinispan_10_0.xsd:1280]: Ah yes. Good catch.

File: clustering/infinispan/extension/src/main/resources/schema/jboss-as-infinispan_10_0.xsd (hunk 3 above)
Comment (pferraro) [reply clustering/infinispan/extension/src/main/resources/schema/jboss-as-infinispan_10_0.xsd:1438]: Good catch.

File: clustering/infinispan/extension/src/main/resources/schema/jboss-as-infinispan_10_0.xsd [hunk 5]
```diff
@@ -0,0 +1,1467 @@
...
+            <xs:enumeration value="2.0"/>
+            <xs:enumeration value="2.1"/>
+            <xs:enumeration value="2.2"/>
+            <xs:enumeration value="2.3"/>
+            <xs:enumeration value="2.4"/>
+            <xs:enumeration value="2.5"/>
+            <xs:enumeration value="2.6"/>
+            <xs:enumeration value="2.7"/>
+            <xs:enumeration value="2.8"/>
+            <xs:enumeration value="2.9"/>
```
[the remark this answers is not in the document]
Comment (pferraro) [reply clustering/infinispan/extension/src/main/resources/schema/jboss-as-infinispan_10_0.xsd:1448]: Good catch.

File: clustering/common/src/main/java/org/jboss/as/clustering/controller/CommonRequirement.java [hunk 6]
```diff
@@ -38,6 +38,7 @@
     ELYTRON("org.wildfly.security.elytron", Void.class),
     LOCAL_TRANSACTION_PROVIDER("org.wildfly.transactions.global-default-local-provider", Void.class),
     MBEAN_SERVER("org.wildfly.management.jmx", MBeanServer.class),
+    MICROPROFILE_CONFIG("org.wildfly.microprofile.config", Void.class),
```
Comment (rhusar) [inline clustering/common/src/main/java/org/jboss/as/clustering/controller/CommonRequirement.java:41]: This doesn't appear to be used anywhere here, perhaps accidental?

**Review by rhusar — CHANGES_REQUESTED** [review]

File: clustering/common/src/main/java/org/jboss/as/clustering/controller/CommonRequirement.java (hunk 6 above)
Comment (pferraro) [reply clustering/common/src/main/java/org/jboss/as/clustering/controller/CommonRequirement.java:41]: Ah yes - that was a remnant from a previous 10.1.x release where MP config was required.  We can remove this now.

**Review by rhusar — APPROVED** [review]
👍  Looks great, I don't see any other issues.

**Comment by bstansberry** [comment]
@asoldano I passed this to you as you know best where we stand on the process aspects of this.

**Comment by pferraro** [comment]
@asoldano There are some fixes to transactional hotrod caches in this branch that I'd like to commit separately to simplify downstream porting.  I'm adding a hold label for now.

**Comment by asoldano** [comment]
> @asoldano I passed this to you as you know best where we stand on the process aspects of this.

@bstansberry the upgrade here is aligned with the plans, so I think we can move forward when Paul is ready with the changes.

**Comment by pferraro** [comment]
I'm adding a hold label while we complete some internal sanity tests.  Will remove when complete.

**Comment by bstansberry** [comment]
retest this please

**Comment by darranl** [comment]
The Windows tests that failed were generally about mixed domain, I don't believe related to Infinispan.

Just kicked off that job again but I don't think they should be an issue.
https://ci.wildfly.org/buildConfiguration/WF_PullRequest_WindowsJdk11/206041?

**Pull request merged**